### ISSUE TITLE
Datastream api service account is not getting created automatically while using project-factory version 16.0.0

### ISSUE THREAD

**Issue by Ranagcp** [terraform-google-modules/terraform-google-project-factory#952]
### TL;DR

We encounterred an issue during project factory version upgrade from 12.0.0 to 16.0.0.
Terraform Apply gets failed with error: Error 400: sevice account service-××××@gcp-sa-datastream.iam.gseviceaccount.com does not exist.
Internally roles/ compute.NetworkUser gets assigned to this account but this account doesnot exists and it produce this error.
Raised case to google support team and they also found same issue and redirected to work with community here . datastream api is enabled but it does not create service account automatically.
We tried creating service account explicitly in the project but it failes with same error during terrafor Apply But when we rerun plan after 10-15 minutes it gets ressolved automatically.

### Expected behavior

If datastreamapi is enabled then assoiciated service account should get created automatically.

### Observed behavior

Datastream service identity is not getting created while using project factory version 14.5.0 or 15.0.0 or 16.0.0.

### Terraform Configuration

```hcl
terraform{
Required_providers{
google = {
Version = ">=4.5"
Source=" hashicorp/google"
}
google_beta = {
Version = ">=4.5"
Source =" hashicorp/google-beta"
}
```

### Terraform Version

```sh
1.6.5
```

**Comment by rafitadiaz** [comment]
I would even ask... Why those users needs to be created?
